### ISSUE TITLE
DataSource Override issue

### ISSUE THREAD

**Issue by user-named-void** [TabularEditor/TabularEditor3#486]
We are building a new BI solution with Analysis Service and Tabular Editor 3.
The model development will be done by multiple persons who all need to connect to the datasources with their own credentials.
In the release pipelines we will change the credentials depending on the environment.

I've found [this](https://docs.tabulareditor.com/te3/features/user-options.html) document stating that it is possible to override the credentials per user. This kinda works, but not completely. The documentation is not very clear how it works and what is needed, but i think i have found the correct settings. But we do have some issues with this option.

- If the user has this option set in his tmuo file, the datasource files still get updated with the users username and a password with only asteriks. This means that every branch and pull request will get changed datasources.

- When only overriding the username the user does not get the popup dialog to enter his password when the model is opened.

- And changing the credentials in the properties window don't change the credentials in the tmuo file, only the datasource file, which we don't want. Seems that only the credential popup window updates the tmuo file. But i cannot find an option to manually open this when the model is open.

The analysis server is running in Azure and it is connecting to an on premise sql server via a gateway.

These are the json files that i use.

tmuo file
```
{
  "UseWorkspace": true,
  "WorkspaceConnection": "provider=MSOLAP;data source=analysisserversource",
  "WorkspaceDatabase": "WorkspaceDB_username",
  "DataSourceOverrides": {
    "DATASOURCE1": {
      "Username": "DOMAIN\\USERNAME"
    },
    "DATASOURCES2": {
      "Username": "DOMAIN\\USERNAME",
      "Password": {
        "Encryption": "UserKey",
        "EncryptedString": "ENCRYPTED_PASSWORD_STRING"
      },
      "PrivacySetting": "Private"
    }
  }
}
```

datasource json
```
{
  "type": "structured",
  "name": "DATASOURCE1",
  "connectionDetails": {
    "protocol": "tds",
    "address": {
      "server": "hostname.domain.com",
      "database": "DATABASENAME"
    }
  },
  "credential": {
    "AuthenticationKind": "UsernamePassword",
    "path": "DATASOURCE1",
    "PrivacySetting": "Private",
    "kind": "SQL",
    "Username": "DOMAIN\\USERNAME",
    "Password": "********"
  }
}
```

**Comment by otykier** [comment]
The asterisk problem may be related to #479. We're looking into it - thanks!

**Comment by otykier** [comment]
This issue should now be fixed in [3.2.3](https://github.com/TabularEditor/TabularEditor3/releases/tag/3.2.3). Thanks for reporting!
